functions/src/functions/webhooks/auth/twilio-verify.ts

```diff
@@ -0,0 +1,108 @@
+import { getAuth } from 'firebase-admin/auth';
+import { getFirestore } from 'firebase-admin/firestore';
+import { HttpsError, onCall } from 'firebase-functions/v2/https';
+import { Twilio } from 'twilio';
+import { TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN, TWILIO_VERIFY_SERVICE_SID } from '../../../config';
+
+if (!TWILIO_ACCOUNT_SID || !TWILIO_AUTH_TOKEN || !TWILIO_VERIFY_SERVICE_SID) {
+	throw new Error('Missing Twilio environment variables');
+}
+
+const twilioClient = new Twilio(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN);
+
+/**
+ * Cloud function to verify OTP sent by Twilio
+ * If valid, it creates/gets a Firebase user and returns a custom token
+ */
+const verifyOtpFunction = onCall({ maxInstances: 10 }, async (request) => {
```

Comment:
Do we need the `maxInstances` parameter here? 

And can you type the request here, i.e. something like

```ts
type VerifyRequest = {
	phoneNumber: string;
	otp: string;
}
```

and then
```ts
onCall<VerifyRequest>()...
```

Reply:
Applied in new PR https://github.com/socialincome-san/public/pull/1119/files